src/main/java/randoop/contract/CompareToAntiSym.java

```diff
@@ -0,0 +1,67 @@
+package randoop.contract;
+
+import randoop.Globals;
+
+/**
+ * The contract: Reversing a compareTo on two objects should result in equal values of opposite signs
+ * <code>Math.signum(x0.compareTo(x1)) == -Math.signum(x1.compareTo(x0))</code>.
+ */
+public class CompareToAntiSym implements ObjectContract {
+    private static final CompareToAntiSym instance = new CompareToAntiSym();
+
+    private CompareToAntiSym() {};
+
+    public static CompareToAntiSym getInstance() {
+        return instance;
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public boolean evaluate(Object... objects) {
+        Object o1 = objects[0];
+        Object o2 = objects[1];
+
+        // If o1 and o2 are comparable objects, check that their signs are opposite of one another
+        if (o1 instanceof Comparable && o2 instanceof Comparable) {
+            Comparable compObj1 = (Comparable)o1;
+            Comparable compObj2 = (Comparable)o2;
+
+            return Math.signum(compObj1.compareTo(compObj2)) == -Math.signum(compObj2.compareTo(compObj1));
+        }
+        // If the compare to operation can't be done, the statement is trivially true
+        return true;
+    }
+
+    @Override
+    public int getArity() {
+        return 2;
+    }
+
+    @Override
+    public String toCommentString() {
+        return "compare-to-anti-symmetry on x0 and x1";
```

Comment:
Please "symmetry" be "symmetric", for consistency with other names of contracts that are adjectives (such as "reflexive").

I would also use "compareTo" instead of "compare-to", since "compareTo" is the Java method being tested.